plugins/catalog/src/components/EntitySwitch/EntitySwitch.tsx

```diff
@@ -64,8 +64,9 @@ export interface EntitySwitchProps {
 
 /** @public */
 export const EntitySwitch = (props: EntitySwitchProps) => {
-  const { entity, loading } = useAsyncEntity();
+  const { entity } = useAsyncEntity();
```

Comment:
Hmm. I'm mildly confused by this.

Everything inside an entity page should be able to use `useEntity()` instead, and rely on it being already ensured to be present, [by way of `EntityLayout`](https://github.com/backstage/backstage/blob/85617f898d72d9ee19e672c2c3f90e2971e67e23/plugins/catalog/src/components/EntityLayout/EntityLayout.tsx#L261).

But for some reason, it seems that `EntitySwitch` has been deemed as needing to have the loading gateway behaviour too. Now we're removing that and hoping that it's safe to do so I guess? Can an entity switch never appear outside of an entity layout? If so, maybe this could be changed to `useEntity` instead, but there are some `if (!entity)` since before below, and I'm kinda wondering, whether that was put in place for a good reason

Reply:
the loading gateway is implicitly still there, as `!entity` happens to be true at first loading time, when no entity has been loaded. But early returning when loading is `true` is also an issue, since we don't want to unmount the entire stack of children on entity refresh.